File: Server/db-migration.js
const { MongoClient } = require("mongodb");

const url = "mongodb://localhost:27017"; // Replace with your MongoDB connection string
const dbName = "your_database_name"; // Replace with your database name

const client = new MongoClient(url, { useNewUrlParser: true, useUnifiedTopology: true });

const createUserTableQuery = [
    {
        $create: "User",
        documents: [
            {
                username: { $type: "string", $unique: true, $required: true },
                password: { $type: "string", $required: true },
                token: { $type: "string", $default: null },
            },
        ],
    },
];

const createTodoTableQuery = [
    {
        $create: "Todo",
        documents: [
            {
                user_id: { $type: "integer", $reference: "User", $required: true },
                title: { $type: "string", $default: "" },
                name: { $type: "string", $required: true },
                completed: { $type: "boolean", $default: false },
            },
        ],
    },
];

async function createTables() {
    try {
        await client.connect();
        const db = client.db(dbName);

        // Create User collection
        await db.createCollection("User");
        await db.collection("User").createIndexes([{ key: { username: 1 }, unique: true }]);

        // Create Todo collection
        await db.createCollection("Todo");
        await db.collection("Todo").createIndexes([{ key: { user_id: 1 } }]);

        console.log('"User" and "Todo" collections created successfully.');
    } catch (err) {
        console.error("Error creating collections:", err);
    } finally {
        await client.close();
    }
}

createTables();

File: Server/db-migration.py
from pymongo import MongoClient, ASCENDING
from pymongo.errors import OperationFailure
from pymongo.collection import Collection

url = "mongodb://localhost:27017"  # Replace with your MongoDB connection string
db_name = "your_database_name"  # Replace with your database name

client = MongoClient(url, connect=False)

create_user_table_query = [
    {
        "$create": "User",
        "documents": [
            {
                "username": {"$type": "string", "$unique": True, "$required": True},
                "password": {"$type": "string", "$required": True},
                "token": {"$type": "string", "$default": None},
            },
        ],
    },
]

create_todo_table_query = [
    {
        "$create": "Todo",
        "documents": [
            {
                "user_id": {"$type": "integer", "$reference": "User", "$required": True},
                "title": {"$type": "string", "$default": ""},
                "name": {"$type": "string", "$required": True},
                "completed": {"$type": "boolean", "$default": False},
            },
        ],
    },
]

def create_tables():
    try:
        client.server_info()
        db = client[db_name]

        # Create User collection
        create_collection(db, "User", create_user_table_query)

        # Create Todo collection
        create_collection(db, "Todo", create_todo_table_query)

        print('"User" and "Todo" collections created successfully.')
    except OperationFailure as e:
        print("Error creating collections:", e)
    finally:
        client.close()

def create_collection(db, collection_name, queries):
    collection = db[collection_name]
    try:
        collection.create_index("username", unique=True)
    except OperationFailure as e:
        print(f"Error creating index on {collection_name}: {e}")

if __name__ == "__main__":
    create_tables()
